src/main/java/com/google/netpcapanalysis/dao/PCAPParserDaoImpl.java

```diff
@@ -0,0 +1,166 @@
+        String dstip = ip.getDestinationIP();
+
+        Buffer buffer = ip.getPayload();
+        int size = buffer.getReadableBytes();
+
+        String protocol = getProtocol(ip); 
+
+        if (first) {
+            MYIP = srcip;
+            first = false;
+        }
+        
+        // PCAPdata takes in (source, destination, domain, location, protocol, size, flagged, frequency) 
+        PCAPdata rawPCAP = new PCAPdata(srcip, dstip, "", "", protocol, size, false, 1);
+        allPCAP.add(rawPCAP);
+      }
+      return true;
+    }
+   });
+   pcap.close();
+  } 
+
+  /* Determine the protocol of IPv4 packet; if unknown, then returns "IPv4" as protocol.
+    * Checks for the following protocols — ICMP, IGMP, TCP, UDP, SCTP, SIP, SDP, ETHERNET_II, SLL, IPv4, RTP —
+    * which are the only protocols supported by pkts library. */
+  private String getProtocol(IPPacket packet) throws IOException {
+    String protocol = "IPv4";
+    // If packet has UDP or TCP protocol, fetch the port numbers (source/destination).
+    if (packet.hasProtocol(Protocol.UDP)) {
+      protocol = "UDP";
+      UDPPacket udpPacket = (UDPPacket) packet.getPacket(Protocol.UDP);
+      int dstport = udpPacket.getDestinationPort();
+      int srcport = udpPacket.getSourcePort();
+    }
+    else if (packet.hasProtocol(Protocol.TCP)) {
+      protocol = "TCP";
+      TCPPacket tcpPacket = (TCPPacket) packet.getPacket(Protocol.TCP);
+      int dstport = tcpPacket.getDestinationPort();
+      int srcport = tcpPacket.getSourcePort();
+    } 
+    else if (packet.hasProtocol(Protocol.ICMP)) {
+      protocol = "ICMP";
+    }
+    else if (packet.hasProtocol(Protocol.IGMP)) {
+      protocol = "IGMP";
+    }
+    else if (packet.hasProtocol(Protocol.SCTP)) {
+      protocol = "SCTP";
+    }
+    else if (packet.hasProtocol(Protocol.SIP)) {
+      protocol = "SIP";
+    }
+    else if (packet.hasProtocol(Protocol.SDP)) {
+      protocol = "SDP";
+    }
+    else if (packet.hasProtocol(Protocol.ETHERNET_II)) {
+      protocol = "ETHERNET_II";
+    }
+    else if (packet.hasProtocol(Protocol.SLL)) {
+      protocol = "SLL";
+    }
+    return protocol;
+  }
+ 
+  /* Removes duplicated data and increments corresponding frequencies */
+  public void processData(){
+    for (PCAPdata packet : allPCAP) {
+      String OUTIP = "";
+      String srcip = packet.source;
+      String dstip = packet.destination;
+
+      if (srcip == MYIP) {
+        OUTIP = dstip;
+      }
+      else {
+        OUTIP = srcip;
+      }
+      
+      // PCAPdata takes in (source, destination, domain, location, protocol, size, flagged, frequency) 
+      if (finalPCAP.containsKey(OUTIP)){
+        // retrieve current value with OUTIP and increments frequency
+        PCAPdata currPCAP = finalPCAP.get(OUTIP);
+        currPCAP.incrementFrequency();
+      }
+      else {
+        PCAPdata tempPCAP = new PCAPdata(MYIP, OUTIP, "", "", packet.protocol, packet.size, packet.flagged, packet.frequency); 
```

Comment:
if this temp is the same as `packet` may be  just add it to `finalPCAP.put(OUTIP, packet)`